lib/pure/rationals.nim

```diff
@@ -46,8 +46,9 @@ func reduce*[T: SomeInteger](x: var Rational[T]) =
     x.num = x.num div common
     x.den = x.den div common
   elif x.den < 0:
-    x.num = -x.num div common
-    x.den = -x.den div common
+    when T isnot SomeUnsignedInt:
```

Comment:
How can `x.den < 0` when typeof(x.den) is an unsigned type?

Reply:
it cant.  but the presence of `-x.den` in this branch is preventing SomeUnsignedInt from compiling.

Reply:
So move the `when` further up, for this the `elif` needs to be turned into an `else`.